src/cli/commands/install.js

```diff
@@ -917,13 +928,8 @@ export function hasWrapper(commander: Object, args: Array<string>): boolean {
 
 export function setFlags(commander: Object) {
   commander.usage('install [flags]');
+  commander.option('--peer', 'explicitly install peer dependencies as well');
   commander.option('-g, --global', 'DEPRECATED');
-  commander.option('-S, --save', 'DEPRECATED - save package to your `dependencies`');
```

Comment:
Why remove these? They are deprecated but removing them is a breaking change and we don't want to release a major version just yet.

Reply:
They aren't deprecated so much as simply invalid. They won't do anything anymore so why report them as deprecated? 

The `--peer` option was going to be re-used in light of this and the other options are only valid in `add`, unless I'm mistaken?

I left `-g` because I think that still makes sense with `install`. Although I think that one is also moot?

Reply:
Removing deprecated features doesn't require a major per semver. I don't have a problem removing them, since it's been pretty clear for quite a long time now that `yarn install` isn't meant to add dependencies.

Reply:
Wait, I thought these options _did not work_ at all with `install`. That's why I removed their listing. If they still function then I stand corrected and they should remain until the feature itself is removed.

(Note that this only removes the `--help` output, not the actual features they once represented...)

Reply:
>Removing deprecated features doesn't require a major per semver. I don't have a problem removing them, since it's been pretty clear for quite a long time now that yarn install isn't meant to add dependencies.

I think this is still a hard-breaking change. I'll ask React guys to see what they do. I think they start deprecating in feature updates and only remove them in major releases which makes sense.

>Wait, I thought these options did not work at all with install. That's why I removed their listing. If they still function then I stand corrected and they should remain until the feature itself is removed.

The feature is always there. These are just legacy flags from NPM but AFAIK they do work.

Reply:
They don't work per-se, but generate a 'suggested command line'.

https://github.com/yarnpkg/yarn/blob/master/src/cli/commands/install.js#L951-L975

@qfox Can you remove their removal from this PR? I think we should focus on the peer dependency itself, removing the deprecated commands will be a separate PR :)